### ISSUE TITLE
Partial template invocation does not accept string literal

### ISSUE THREAD

**Issue by dankolz** [wycats/handlebars.js#1013]
I get this error in version 1.3.0 when trying to invoke a partial template with a string. I've modified the error to remove the specific template name and string I'm passing.

Error: Parse error on line 25:...partialTemplateName 'string literal'}-----------------------^Expecting 'CLOSE', 'ID', got 'STRING'

What makes sense to me is to make the string accessible via {{this}} in the partial template. It seems that the parser code is already identifying the condition. Can this be easily done, do you know? Is it a design decision not to allow it? Would you take a patch for it?

**Comment by kpdecker** [comment]
I'm not sure what you are asking here. Can you provide a code example?

**Comment by dankolz** [comment]
You bet. The code is something like

{{> i18n-message 'my.message.key'}}

Where the partial template 18n-message I want to look like

<div class="translated-message">
{{i18n-lookup this}}
</div>

What I wanted to do is have the i18n-message partial template provide markup and the i18n-lookup helper do the actual conversion from message key to message in the locale.

The part which fails with the parse exception is:

{{> i18n-message 'my.message.key'}}

**Comment by dankolz** [comment]
hmm. Github filtered my markup. Let me try again.

**Comment by dankolz** [comment]
![image](https://cloud.githubusercontent.com/assets/1598367/7395066/4cfa5d42-ee5d-11e4-8f57-35b685c3e80e.png)

**Comment by dankolz** [comment]
The real markup is a bit more complicated. I've reduced it for readability here. I don't think that should matter though since it's the parsing of the invocation 

{{> i18n-message 'my.message.key'}}

where I'm getting the parse error.

**Comment by kpdecker** [comment]
I'll have to look into the ramifications of passing a non-object as the partial context to see what the impact might be. In the immediate term, you can use partial parameters to get around this:

```
{{> i18n-message key='my.message.key' }}
```

```
<div>{{i18n-lookup key}}
```

**Comment by kpdecker** [comment]
This actually works in later versions, although I'm not sure when it was introduced. I'd recommend updating to the latest released version. I've added explicit tests to ensure that this isn't lost in the future.
